### PR TITLE
Feature/social login signup

### PR DESCRIPTION
**Pull request by ruihildt**
# Description
Add Github signup/login endpoint.

## Checklist
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by iamcristos** [comment]
your build is failing did you pull from development before commiting

File: api/middleware/authStrategies.js [hunk 1]
```diff
@@ -7,16 +7,51 @@ const TwitterStrategy = require('passport-twitter').Strategy;
...
-async function callbackStrategy(profile, cb) {
-  const userEmail = await models.Users.findOne({
-    where: { email: profile.email }
+passport.serializeUser((user, done) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser((id, done) => {
+  models.Users.findByPk(id, (err, user) => {
+    console.log('deserialize', user);
```
Comment (iamcristos) [inline api/middleware/authStrategies.js]: can you take out this console.log statement

File: api/middleware/authStrategies.js [hunk 2]
```diff
@@ -7,16 +7,51 @@ const TwitterStrategy = require('passport-twitter').Strategy;
...
+        defaults: { username: profile.username, email, password: ' ' }
+      });
+      if (!newUser) {
+        return new Error();
+      }
+      return cb(null, newUser);
+    }
+    return cb(null, existingUser);
+  } catch (error) {
+    console.log(error.message);
```
Comment (iamcristos) [inline api/middleware/authStrategies.js]: here also

File: api/middleware/authStrategies.js (hunk 1 above)
Comment (ruihildt) [reply api/middleware/authStrategies.js]: Fixed.

File: api/middleware/authStrategies.js (hunk 2 above)
Comment (ruihildt) [reply api/middleware/authStrategies.js]: Fixed.

**Pull request merged**